idea/idea-fir/src/org/jetbrains/kotlin/idea/fir/intentions/HLConvertToBlockBodyIntention.kt

```diff
@@ -104,9 +100,8 @@ class HLConvertToBlockBodyIntention :
         private fun generateBody(body: KtExpression, input: Input, returnsValue: Boolean): KtExpression {
             val factory = KtPsiFactory(body)
             if (input.bodyTypeIsUnit && body is KtNameReferenceExpression) return factory.createEmptyBody()
-            val unitWhenAsResult = input.bodyTypeIsUnit && body.resultingWhens().isNotEmpty()
```

Comment:
You can delete the `resultingWhens()` function you added.

Reply:
I didn't delete in commit#4 since it's also used in `UseExpressionBodyInspectioin`. So when we rewrite that, we probably would find that useful?

Reply:
Oh okay, I didn't realize you had just moved it, I thought you added it.